swatch-core/src/main/java/org/candlepin/subscriptions/db/BillableUsageRemittanceFilter.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright Red Hat, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Red Hat trademarks are not licensed under GPLv3. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.subscriptions.db;
+
+import java.time.OffsetDateTime;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Objects;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.Setter;
+import org.candlepin.subscriptions.db.model.BillableUsageRemittanceEntityPK_;
+import org.candlepin.subscriptions.db.model.BillableUsageRemittanceEntity_;
+
+/**
+ * A filter used to find {@link org.candlepin.subscriptions.db.model.BillableUsageRemittanceEntity}
+ * objects via the {@link BillableUsageRemittanceRepository}. Any filter value with a null value
+ * will not be checked.
+ */
+@Builder
+@Getter
+@Setter
+public class BillableUsageRemittanceFilter {
```

Comment:
I'd rather see this class implement Specification. That'll remove a layer of abstraction and reduce number of classes needed. It should more-or-less mean consolidating `getSearchCriteria` and `toSpecification`.

Reply:
Made the requested changes. Let me know if it is what you had in mind.

Reply:
Looks good, thanks!